napari/utils/colormaps/colormaps.py

```diff
@@ -111,6 +117,31 @@ def _low_discrepancy_image(image, seed=0.5, margin=1 / 256):
     return image_out
 
 
+def color_dict_to_colormap(color_dict):
+    """
+    Generate a color map based on the given color dictionary
+    Parameters
+    ----------
+    color_dict : dict of label and color string matches
+
+    Returns
+    -------
+    colormap : Colormap with provided control colors
+    custom_color : dict of label to color control point within colormap
+    """
+    custom_color = {0: 0.0}
+    colors = ['#000000'] + [
+        color_str for label, color_str in color_dict.items()
+    ]
+    colormap = Colormap(colors)
+    for label, color_str in color_dict.items():
+        color = Color(color_str)
+        for i in range(len(colormap.colors)):
+            if colormap.colors[i] == color:
+                custom_color[label] = colormap._controls[i]
+    return colormap, custom_color
```

Comment:
Can we rename `custom_color` to `label_color_index` as it is sort of analogous to how we use `_label_index` in the labels layer